src/licenses/LicenseFilter.cpp

```diff
@@ -0,0 +1,46 @@
+#include "LicenseFilter.h"
+
+#include <QtCore/QFile>
+#include <QTextStream>
+
+namespace licenses {
+
+  LicenseFilter::LicenseFilter(QObject *parent) : QSortFilterProxyModel(parent) {}
+
+  bool LicenseFilter::filterAcceptsRow(int source_row, const QModelIndex &source_parent) const {
+    QModelIndex index = sourceModel()->index(source_row, 0, source_parent);
+
+    auto categories = sourceModel()->data(index, LicenseCategories).toList();
+
+    bool support = false;
+
+    for (const auto &category : categories) {
+      auto cat = static_cast<LicenseCategory>(category.toInt());
+      if (cat == m_category)
+        support = true;
+    }
```

Comment:
isn't this basically the same as `auto support = categories.contains(static_cast<int>(m_category))`?
From a purely academic standpoint, it's also better to cast the enum to the broader int instead of the other way round, since the QVariant in the list might hold a value that is not a valid `LicenseCategory` 🤓

Reply:
Oooh, and much prettier!